src/System.Text.JsonLab/System/Text/Json/JsonWriterUtf16.cs

```diff
@@ -570,10 +570,17 @@ public void WriteNull()
                 }
             }
 
-            Debug.Assert(byteBuffer.Slice(idx * 2).Length >= JsonConstants.NullValueUtf16.Length);
-            JsonConstants.NullValueUtf16.CopyTo(byteBuffer.Slice(idx * 2));
-
-            _bufferWriter.Advance(bytesNeeded);
+            if (BitConverter.IsLittleEndian)
+            {
+                Debug.Assert(byteBuffer.Slice(idx * 2).Length >= JsonConstants.NullValueUtf16LE.Length);
+                JsonConstants.NullValueUtf16LE.CopyTo(byteBuffer.Slice(idx * 2));
+                _bufferWriter.Advance(bytesNeeded);
+            }
+            else
+            {
+                _bufferWriter.Advance(bytesNeeded);
```

Comment:
Add a NullValueUtf16BE and use that instead?

Alternatively, try to isolate and measure the difference in performance between calling the UTF-16 to UTF-8 encode method for a small input (like 4-byte null array) and compare that to directly copying a pre-computed byte array. If there is little performance impact, we could keep the encoding step, which appropriately deals with the endianness.